src/assertions/object_type.h

```diff
@@ -0,0 +1,72 @@
+#include <xboxkrnl/xboxkrnl.h>
+
+#include "util/output.h"
+#include "defines.h"
+
+static BOOL assert_object_type_ex(POBJECT_TYPE object_type,
+                                  ULONG pool_tag,
+                                  BOOL has_close,
+                                  BOOL has_delete,
+                                  BOOL has_parse,
+                                  int func_line)
```

Comment:
Instead of passing all the arguments to check to the function like this, you could create an array in `src/tests/io/suite/object_type.c` that holds all of them, and then this function would only receive an index/enumerator to the object to check.
```
struct object_info {
	POBJECT_TYPE object_type;
	ULONG pool_tag;
	BOOL has_close;
	BOOL has_delete;
	BOOL has_parse;
};

enum object_t {
	event_object_type = 0,
	// all the others
	object_max
};

// This could also use designated initializers instead of comments
object_info objects_to_test_info[object_max] = {
	{
		/* object_type */ &ExEventObjectType,
		/* pool_tag */ 'vevE',
		/* has_close */ FALSE,
		/* has_delete */ FALSE,
		/* has_parse */ FALSE
	}
};
```

Reply:
An array would centralize the tested characteristics of all object types into one single place, which sentiment could be appreciated, but having to refer to them by index (while there's no other usage of such an index) introduces an extra indirection step that not only obfuscates the actual values used for the calls, but also requires careful maintenance (if anything would ever be changed, that is).

All in all, I'd go with the argument approach as it is right now in this pull request.